src/system_metrics/mongoose_system_metrics_collector.erl

```diff
@@ -33,10 +34,55 @@ get_hosts_count() ->
 
 get_modules() ->
     Hosts = ejabberd_config:get_global_option(hosts),
+    AllModules = lists:flatten(
+                    lists:map(fun gen_mod:loaded_modules/1, Hosts)),
+    ModulesToReport = filter_behaviour_implementations(AllModules,
+                                                       mongoose_module_metrics),
     ModulesWithOpts = lists:flatten(
-        lists:map(fun gen_mod:loaded_modules_with_opts/1, Hosts)),
+                        lists:map(
+                            fun(Host) ->
+                                get_modules_metrics(Host, ModulesToReport)
+                            end, Hosts)),
     lists:map(
         fun({Module, Opts}) ->
-            Backend = proplists:get_value(backend, Opts, none),
-            #{report_name => Module, key => backend, value => Backend}
+            report_module_with_opts(Module, Opts)
         end, ModulesWithOpts).
+
+filter_behaviour_implementations(Modules, Behaviour) ->
+    lists:filter(
+        fun(M) ->
+             try lists:keyfind([Behaviour], 2, M:module_info(attributes)) of
+                 {behavior, _} -> true;
+                 {behaviour, _} -> true;
+                 _ -> false
+             catch
+                 _:_ -> false
+             end
+         end, Modules).
+
+get_modules_metrics(Host, Modules) ->
+    lists:map(
+        fun(M) ->
+            case erlang:function_exported(M, config_metrics, 1) of
+                true -> {M, M:config_metrics(Host)};
+                false -> {M ,[{none, none}]}
+            end
+        end, Modules).
+
+report_module_with_opts(Module, Opts) ->
+    lists:map(
+        fun({OptKey, OptValue}) ->
+            #{report_name => Module, key => OptKey, value => OptValue}
+        end,Opts).
+
+get_number_of_custom_modules() ->
+    Hosts = ejabberd_config:get_global_option(hosts),
+    AllModules = lists:flatten(
+                    lists:map(fun gen_mod:loaded_modules/1, Hosts)),
+    GenMods = filter_behaviour_implementations(AllModules, gen_mod),
+    GenModsSet = sets:from_list(GenMods),
+    MetricsModule = filter_behaviour_implementations(AllModules,
+                                                     mongoose_module_metrics),
+    MetricsModuleSet = sets:from_list(MetricsModule),
+    CountCustomMods= sets:size(sets:subtract(GenModsSet, MetricsModuleSet)),
+    #{report_name => custom_modules_names, key => sets:to_list(sets:subtract(GenModsSet, MetricsModuleSet)), value => CountCustomMods}.
```

Comment:
😱

Reply:
work in progress, the line will be different after testing is copleted